### ISSUE TITLE
2.0.1: Memory leak when an element in a component has a handlebars'd attribute

### ISSUE THREAD

**Issue by arthare** [emberjs/ember.js#12265]
This appears to have been present at least as far back as 1.13.5.  It is separate from the other memory leak I mentioned earlier today that was fixed in 1.13.4.

Sample app: https://github.com/arthare/leakyEmberApp
Repro instructions are in the readme.md in that github.

Based on my research, it looks like whenever a DOM element in a component has an attribute made using a handlebars expression, the entire component gets leaked on route transitions.

Example: This will leak [inside a component's template]:
```
<span fake-attrib="leaky-{{product.id}}">The parent span of this text might be leaky!</span>
```
This will not [inside a component's template]:
```
<span fake-attrib="leaky-5">The parent span of this text might be leaky!</span>
```

Other tidbits:
* The element with the handlebars'd attribute has to be inside a component.  I tried putting the same span in my route, and the leaks didn't happen.
* The amount leaked is proportional to the element count of the component, but is not affected by the element count of the containing route.  This seems to indicate the rendered component is being leaked, but not the route.
* I first noticed this in my 1.13.5-based app, and repro'd it up to 1.13.9 in that app, and have repro'd it in 1.13.5, 1.13.9, and 2.0.1 in the toy app.

**Comment by stefanpenner** [comment]
can you try master?

**Comment by raytiley** [comment]
I think this might be fixed with https://github.com/emberjs/ember.js/pull/12256

There was a leak in `concat` which is used to combine the strings.

**Comment by arthare** [comment]
That sounds exactly like it.  Will give it a try.

**Comment by rwjblue** [comment]
Thanks for the thorough bug report, please confirm with latest builds that it is indeed fixed.  You should be able to use latest builds from any channel to get the fix from #12256.

**Comment by arthare** [comment]
That did it.  Thanks everyone.  What a coincidence that it got PR'ed today just as I narrowed it down on my end.
